notebooks/try-annoy.py

```diff
@@ -0,0 +1,401 @@
+        sns,
+    )
+
+
+@app.cell
+def _(mo):
+    mo.md(
+        r"""
+    # Test using Annoy to find similar sentences
+    In this notebook we will will find the the top 10 nearest MEGA sentences for each DNZ sentence.
+    In other words, the MEGA sentences will form the index and the DNZ sentences will be queries.
+    """
+    )
+    return
+
+
+@app.cell
+def _(np, pl):
+    # Load sentence corpora and embeddings
+    ## Load sentence corpora
+    dnz_sents_df = (
+        pl.scan_ndjson("data/sentence-corpora/dnz-sample-sents.jsonl")
+        .select(["sent_id", "text"])
+        .with_row_index()
+        .rename(lambda x: f"dnz_{x}")
+        .collect()
+    )
+    marx_sents_df = (
+        pl.scan_ndjson("data/sentence-corpora/mega-sample-sents.jsonl")
+        .select(["sent_id", "text"])
+        .with_row_index()
+        .rename(lambda x: f"marx_{x}")
+        .collect()
+    )
+
+    ## Load embeddings
+    marx_embeddings = np.load("data/sentence-embeddings/mega-sample-sents.npy")
+    dnz_embeddings = np.load("data/sentence-embeddings/dnz-sample-sents.npy")
+    return dnz_embeddings, dnz_sents_df, marx_embeddings, marx_sents_df
+
+
+@app.cell
+def _(mo):
+    mo.md(r"""## Run Annoy""")
+    return
+
+
+@app.cell
+def _(mo):
+    mo.md(
+        r"""
+    For our Annoy index, we will use the dot product as our distance measurement (note that it really is a similarity measurement and is treated as such by Annoy).
+    In our case is the same as cosine similarity since our embeddings are L2 normalized.
+
+    An Annoy index takes one hyperparameter which specifies the number of trees the index will construct. More trees means better precision, but requires more resources.
+    For now, we will arbitrarily use 10 trees.
+
+    It's worth noting that we can save the constructed Annoy index over MARX sentences to disk and reload it later to compare to any additional sentence embeddings we might have later on.
+    This could be a possible advantage over other methods.
+    """
+    )
+    return
+
+
+@app.cell
+def _(AnnoyIndex, marx_embeddings):
+    # Build annoy index for Marx sentence embeddings
+    n_features = marx_embeddings.shape[1]
+
+    # Instantiate annoy index, use dot product
+    marx_index = AnnoyIndex(n_features, "dot")
+
+    # Add each embedding to index
+    for i, row in enumerate(marx_embeddings):
+        marx_index.add_item(i, row)
+
+    # Build index with 10 trees, for now not saving (will do this if building the index is too slow)
+    _ = marx_index.build(10)
+    return (marx_index,)
+
+
+@app.cell
+def _(dnz_embeddings, dnz_sents_df, marx_index, marx_sents_df, pl):
+    # Build knn results
+    entries = []
+    for dnz_i, dnz_vec in enumerate(dnz_embeddings):
+        knn_results = marx_index.get_nns_by_vector(
+            dnz_vec, n=10, include_distances=True
+        )
+        for rank in range(10):
+            # Marx sentence row index
+            marx_i = knn_results[0][rank]
+
```

Comment:
I'm getting an `IndexError` on line 129. I tried inspecting and it seems that I'm only getting one result per query, rather than the expected 10. Any ideas?